src/bmi_troute.py

```diff
@@ -0,0 +1,631 @@
+        'usgs_gage_observation__volume_flow_rate':['streamflow_cms','m3 s-1'],
+        'usace_gage_observation__volume_flow_rate':['streamflow_cms','m3 s-1'],
+        'rfc_gage_observation__volume_flow_rate':['streamflow_cms','m3 s-1']
+    }
+
+    #------------------------------------------------------
+    # A list of static attributes. Not all these need to be used.
+    #------------------------------------------------------
+    _static_attributes_list = []
+
+
+    #------------------------------------------------------------
+    #------------------------------------------------------------
+    # BMI: Model Control Functions
+    #------------------------------------------------------------ 
+    #------------------------------------------------------------
+
+    #-------------------------------------------------------------------
+    def initialize(self, bmi_cfg_file=None):
+
+        args = tr._handle_args_v03(['-f', bmi_cfg_file])
+        
+        # -------------- Read in the BMI configuration -------------------------#
+        bmi_cfg_file = Path(bmi_cfg_file)
+        # ----- Create some lookup tabels from the long variable names --------#
+        self._var_name_map_long_first = {long_name:self._var_name_units_map[long_name][0] for \
+                                         long_name in self._var_name_units_map.keys()}
+        self._var_name_map_short_first = {self._var_name_units_map[long_name][0]:long_name for \
+                                          long_name in self._var_name_units_map.keys()}
+        self._var_units_map = {long_name:self._var_name_units_map[long_name][1] for \
+                                          long_name in self._var_name_units_map.keys()}
+        
+        # This will direct all the next moves.
+        if bmi_cfg_file is not None:
+
+            with bmi_cfg_file.open('r') as fp:
+                cfg = yaml.safe_load(fp)
+            self._cfg_bmi = self._parse_config(cfg)
+        else:
+            print("Error: No configuration provided, nothing to do...")
+        
+        # ------------- Initialize t-route model ------------------------------#
+        (self._network, 
+         self._log_parameters, 
+         self._preprocessing_parameters, 
+         self._supernetwork_parameters, 
+         self._waterbody_parameters, 
+         self._compute_parameters, 
+         self._forcing_parameters, 
+         self._restart_parameters, 
+         self._hybrid_parameters, 
+         self._output_parameters, 
+         self._parity_parameters, 
+         self._data_assimilation_parameters,
+         self._run_parameters,
+        ) = tr.initialize_network(args)
+        
+        # Set number of time steps (1 hour)
+        self._nts = 12
+        
+        # -------------- Initalize all the variables --------------------------# 
+        # -------------- so that they'll be picked up with the get functions --#
+        for var_name in list(self._var_name_units_map.keys()):
```

Comment:
Creating a list here is unnecessary. The keys can be iterated thru directly.
`for var_name in self._var_nname_units_map:`